src/ui/Quickview/Quickview.ts

```diff
@@ -378,29 +378,49 @@ export class Quickview extends Component {
     };
   }
 
-  private prepareQuickviewContent(loadingAnimation: HTMLElement | Promise<HTMLElement>): Promise<Dom> {
-    return this.options.contentTemplate.instantiateToElement(this.result).then((built: HTMLElement) => {
-      const content = $$(built);
-
-      const initOptions = this.searchInterface.options;
-      const initParameters: IInitializationParameters = {
-        options: initOptions,
-        bindings: this.getBindings(),
-        result: this.result
-      };
-      return Initialization.automaticallyCreateComponentsInside(content.el, initParameters).initResult.then(() => {
-        if (content.find('.' + Component.computeCssClassName(QuickviewDocument)) != undefined && this.options.enableLoadingAnimation) {
-          if (loadingAnimation instanceof HTMLElement) {
-            content.prepend(loadingAnimation);
-          } else if (loadingAnimation instanceof Promise) {
-            loadingAnimation.then(anim => {
-              content.prepend(anim);
-            });
-          }
-        }
-        return content;
-      });
-    });
+  private async prepareQuickviewContent(loadingAnimation: HTMLElement | Promise<HTMLElement>): Promise<Dom> {
+    const domContent = await this.instantiateTemplateToDom();
+
+    const initOptions = this.searchInterface.options;
+    const initParameters: IInitializationParameters = {
+      options: initOptions,
+      bindings: this.getBindings(),
+      result: this.result
+    };
+
+    await Initialization.automaticallyCreateComponentsInside(domContent.el, initParameters).initResult;
+
+    const containsQuickviewDocumentAndCustomAnimation = () =>
+      domContent.find(`.${Component.computeCssClassName(QuickviewDocument)}`) != undefined && this.options.enableLoadingAnimation;
+
+    if (containsQuickviewDocumentAndCustomAnimation()) {
+      if (loadingAnimation instanceof HTMLElement) {
+        domContent.prepend(loadingAnimation);
+      } else if (loadingAnimation instanceof Promise) {
+        loadingAnimation.then(anim => {
+          domContent.prepend(anim);
+        });
+      }
+    }
+    return domContent;
+  }
+
+  private async instantiateTemplateToDom(): Promise<Dom> {
```

Comment:
Would it be possible to simplify by removing the `finally` and moving the logic to `catch`:
```
private async instantiateTemplateToDom(): Promise<Dom> {
    let el: HtmlElement;
    try {
        el = await this.options.contentTemplate.instantiateToElement(this.result);
    } catch {
        // log message
        el = await new DefaultQuickviewTemplate().instantiateToElement(this.result);
    }
    return $$(el);
}
```

unless the call in the `try` block can return `null/undefined`, in which case you will need to check and throw an error to trigger the `catch` block.

Reply:
Yes it can return undefined or null (that's the original bug in the ticket), and in that case, I would indeed need to throw an error.

It seems to me it's more natural to do the null/undefined check in the finally block, instead of throwing an error, otherwise for someone reading this, it might look like we are only throwing an error to check for null/undefined in a weird way, and this "hide" the fact that `instantiateToElement` itself can also throw an error.